js/src/table.ts

```diff
@@ -238,17 +238,30 @@ export class Table<T extends TypeMap = any> {
      *
      * @returns An Array of Table rows.
      */
-    public toArray() {
+    public toArray(): Array<Struct<T>['TValue']> {
         return [...this];
     }
 
+    /**
+     * Return a JavaScript Array view of the Table rows.
+     *
+     * It is a lightweight read-only proxy that delegates to the table. Accessing elements has some
+     * overhead compared to the regular array returned by `toArray()` because of this indirection,
+     * but it avoids potentially large memory allocation.
+     *
+     * @returns An Array proxy to the Table rows.
+     */
+    public toArrayView(): Array<Struct<T>['TValue']> {
+        return new Proxy([] as Array<Struct<T>['TValue']>, new TableArrayProxyHandler(this));
+    }
+
     /**
      * Returns a string representation of the Table rows.
      *
      * @returns A string representation of the Table rows.
      */
     public toString() {
-        return `[\n  ${this.toArray().join(',\n  ')}\n]`;
+        return `[\n  ${this.toArrayView().join(',\n  ')}\n]`;
```

Comment:
Since we are stringifying the whole table, is it worth using an array view here?

Reply:
We need an array (or array-like) to use `join()`. This _was_ creating an array, and with `toArrayView()` this now just creates a proxy. Or did I misunderstand your comment?

Reply:
I understand the difference but I wonder whether this change introduces a regression since "the proxy adds some overhead to direct array access."

Reply:
Got it. I don't think this is an issue here, as `toString()` is generally used for debugging purposes. Actually I'm wondering if it's practically usable for `Table` beyond tests as it can create huge strings.